lib/contextual_logger.rb

```diff
@@ -1,26 +1,28 @@
 # frozen_string_literal: true
 
 require 'json'
+require_relative './contextual_logger/context/handler'
 
 module ContextualLogger
   def self.new(logger)
     logger.extend(self)
   end
 
   def global_context=(context)
-    Thread.current[THREAD_CONTEXT_NAMESPACE] = context
+    ContextualLogger::Context::Handler.new(context).set!
   end
 
   def with_context(context)
-    previous_context = Thread.current[THREAD_CONTEXT_NAMESPACE] || {}
-    Thread.current[THREAD_CONTEXT_NAMESPACE] = previous_context.merge(context)
+    context_handler = ContextualLogger::Context::Handler.new(current_context_for_thread.merge(context))
+    context_handler.set!
     yield if block_given?
+    context_handler
```

Comment:
is there any benefit to returning the context handler here? It breaks the return of anything that happens in the block provided.

For instance, if there is any code that currently does
```rb
return_value = Logger.with_context(context: "logging") { Model.create! }
```
we will no longer get the created model as the response but instead will get the context handler

Reply:
good catch, definitely want to only return if no block was given. will update that